pandas/tests/reshape/merge/test_merge.py

```diff
@@ -1064,11 +1084,27 @@ def test_merge_right_index_right(self):
                                  'key': [0, 1, 1, 2],
                                  'b': [1, 2, 2, 3]},
                                 columns=['a', 'key', 'b'],
-                                index=[0, 1, 2, 2])
+                                index=[0, 1, 2, np.nan])
```

Comment:
why did this change?

Reply:
The original behaviour of trying to infer the index value does not always work when `other_index` has a different dtype (for instance, if `index` is a `DateTimeIndex` and `other_index` is an `IntervalIndex` then `index` can't take the values stored in `other_index` nor can it take the row number, so the only solution is to use NA values). Keeping the old behaviour would mean having to dispatch on the dtypes of both indices.